### ISSUE TITLE
Analysis: Reviewing Expected Branch Structure and Workflow

### ISSUE THREAD

**Issue by BatemanVO** [aggiewb/salmon-cookies#7]
Analysis tickets do not contain any code or have any items for deployment, but are typically a precursor to another ticket that require determining some information necessary to know what the best solution would be or how difficult an ask might be.

As part of onboarding, we'll be reviewing proper branch structure, naming conventions, and deployment workflow before beginning work on #5.

The workflow should be:
- `git checkout master`
- `git pull origin master` (to ensure you are about to create a branch that has the latest production code)
- `git checkout -b feature/issue number_description`, e.g., `git checkout -b feature/5-replace_object_literals_with_constructor_function`
- Complete work in local dev environment on the feature branch, committing regularly
- When work is complete, open a PR from the feature branch to `staging`
- Code review is performed, feedback is responded to or acknowledged
- Pull request is approved, feature branch is merged to `staging` _but not deleted_
- QA/UAT is performed on tickets that are in `staging` and marked as ready for QA/UAT
- Once QA/UAT is complete, ticket is moved to Ready For Deployment
- A release branch will be made from `master` by the release manager. Devs should merge their feature branch to the release branch (yes, this means the feature branch has been merged to two branches)
- Once all stories ready for release are merged to the release branch, a pull request is opened to `master`
- All devs with stories deploying will review the `release` -> `master` pull request and confirm that all expected code items are present, approving the pull request as they do
- Once all devs have confirmed the `release` branch contains all expected changes, `release` is merged to `master`
- Release branch is deleted, as are all feature branches that no longer have commits that aren't in `master`
